File: youqugou-interface/src/main/java/cn/itcast/core/service/AddressService.java
package cn.itcast.core.service;

import cn.itcast.core.pojo.address.Address;
import cn.itcast.core.pojo.address.Areas;
import cn.itcast.core.pojo.address.Cities;
import cn.itcast.core.pojo.address.Provinces;

import java.util.List;

public interface AddressService {
    List<Address> findListByLoginUser(String username);

    void addressService(Address address);

    void deleteAddress(Long id);

    void update(Address address);

    Address findOne(Long id);

   /* List<Provinces> findList();

    List<Cities> findList2(String id);

    List<Areas> findList3(String id);*/
}

File: youqugou-web-user/src/main/webapp/js/service/addressService.js
app.service("addressService", function ($http) {

    /**
     * 马超  2019.1.2
     * @param entity
     * @returns {*|void}
     */
    // this.addAddress = function (entity) {
    //     return $http.post("../address/addAddress.do",entity)
    // }
    //增加
    this.add=function(entity){
        return  $http.post('../address/addAddress.do',entity );
    }
    //修改
    this.update=function(entity){
        return  $http.post('../address/update.do',entity );
    }

    this.loginInfo=function () {
        return $http.post("../address/loginInfo.do")
    }
    //删除
    this.deleteAddress=function (id) {
        return $http.post("../address/deleteAddress.do",id)
    }

    // //编辑地址回显
    // this.editBack=function (id) {
    //     return $http.post("../address/editBack.do",id)
    // }

    this.findOne=function (id) {
        return $http.get('../address/findOne.do?id='+id);
    }

  /*  this.findList=function () {
        return $http.get('../address/findList.do');
    }

    this.findList2=function (id) {
        return $http.get('../address/findList2.do?id='+id);
    }

    this.findList3=function (id) {
        return $http.get('../address/findList3.do?id='+id);
    }*/

});